### DISCUSSION TITLE
dynamic route + endpoint serves endpoint instead of page

### DISCUSSION THREAD

**Discussion by ITwrx** [sveltejs/kit#4162] [Q&A] [answered]
I'm using v288. Can anyone else confirm this behavior or am i doing something wrong?

i have:
index.svelte
index.js
[root_page].svelte
[root_page].js
[section] (directory)
-- [section_page].svelte
-- [section_page].js

index (http://localhost:3000) loads properly every time.
[root_page].svelte (for dynamic pages at the root dir level. excluding index. i.e http://localhost:3000/about-page) always returns json like it's trying to use the endpoint ( [root_page].js ) as the page.
[section]/[section_page].svelte (http://localhost:3000/software/superapp) renders correctly as html on the first page load, and then as json upon refresh (everytime).

[root_page].svelte
```
<script>	
	export let sections, slug_page;  
</script>

<div class="text-center">
    {#each sections || [] as section}    
        {#if section.inTopNav == 1}                       
            <a href="/{section.slug}" class="ml-4">{section.title}</a>                       
        {/if}  
    {/each}    
</div>
<br><br>
<h1>{slug_page.title}</h1>
```
[root_page].js
```
export async function get({params}) {
  
    let sections = [];  
    const sections_endpoint = "http://127.0.0.1:8000/api/sections";
    const sections_res = await fetch(sections_endpoint);
    sections = await sections_res.json();

    let slug_page = [];  
    const slug_page_endpoint = `http://127.0.0.1:8000/api/active-page-by-slug/${params.root_page}`;
    const slug_page_res = await fetch(slug_page_endpoint);
    slug_page = await slug_page_res.json();
  
    return {
      status: 200,
      body: { sections, slug_page } 
    };
}
```
index and section_pages are more of the same, with index not having a slug param. Manually passing in the string for the slug for the [root_page] has no effect. Same behavior. 

Thanks

**Comment by geoffrich** [comment]
Sounds like this might be the same issue as #4038.

**Reply by ITwrx** [reply]
thanks for your reply. i'm guessing it has to do with recent changes and probably related to that issue, but shouldn't my `index.svelte` and my `[root_page].svelte` not be considered clashing? there has to be a way to designate both `index only` and `anything else at same level`, right? or i'm doing this wrong? or it's currently not supported/implemented?

**Comment by JSaretin** [comment]
Most of your routes are dynamic, I would advise you to prefix the route with a static word like `about-[something].svelte`, adding the same name  .js and .svelte do not populate your .svelte with the result from the .js as stated [here](https://kit.svelte.dev/docs/routing#endpoints). Instead, the .js with override the .svelte. The way to get around this is to create the same route in `/api/about-[something].js ` (might also be in the same directory) and fetch the date from your load function.

**Comment by ITwrx** [comment, marked as answer]
@JSaretin With `shadow endpoints`, now renamed to `endpoints` the `.svelte` file _**is**_ populated with the data from the same-named `.js`, as described in the updated/current docs. The example `.svelte` and `.js` are both named the same, and the file names are dynamic, though not at the top level directory. 

My routes are fully dynamic because i'm trying to build a CMS with Sveltekit, where the admin user needs to have full control over the structure of the resulting public-facing site. For SEO, clarity, and aesthetic reasons, i can't really be putting words in _their_ routes like `category-[the-cat-you-really-wanted].svelte`. Also, my `.svelte` files don't have `load` functions anymore since the change to `shadow endpoints` (now just called `endpoints`, as mentioned above). 

I believe my fully dynamic routes didn't work due to a bug (or just unfinished implementation), which was fixed in sveltekit 293 or 294. They work now (it serves the same-named .`svelte` instead of directly serving the JSON result from the `.js` file) . 

I should have reported back and closed this discussion, so you didn't have to spend your time trying to help. Thanks though!

**Reply by JSaretin** [reply]
Thank you so much for this response, I didn't know that the issue has be resolved, which gave me a lot of headaches when I was trying to get it to work, this will save me a lot of time. thanks once again.

**Reply by ITwrx** [reply]
no problem! yeah, it was making me think i was doing something wrong. :)
